src/Microsoft.AspNetCore.Mvc.Modules/ModularApplicationPart.cs

```diff
@@ -74,18 +72,18 @@ public IEnumerable<TypeInfo> Types
         {
             get
             {
-                return DefaultAssemblyDiscoveryProvider
-                    .DiscoverAssemblies(
-                        ShellBlueprint.Dependencies.Select(dep => dep.Type.GetTypeInfo().Assembly).Distinct().ToList(),
-                        ReferenceAssemblies)
+                return DefaultModularAssemblyDiscoveryProvider
+                    .GetCandidateAssemblies(ExtensionLibraryService.LoadedAssemblies, ReferenceAssemblies)
                     .SelectMany(x => x.DefinedTypes);
             }
         }
 
         /// <inheritdoc />
         public IEnumerable<string> GetReferencePaths()
         {
-            return AssemblyProvider.GetAssemblies().Select(x => x.Location);
+            return DependencyContext.Default.CompileLibraries
+                .SelectMany(library => library.ResolveReferencePaths())
+                .Union(ExtensionLibraryService.MetadataPaths);
```

Comment:
Yes, through the `ICompilationReferencesProvider` which is now used by razor compilation.